cli/cli-client/src/main/java/org/infinispan/cli/user/UserTool.java

```diff
@@ -0,0 +1,252 @@
+         ScramDigestPassword.ALGORITHM_SCRAM_SHA_1,
+         ScramDigestPassword.ALGORITHM_SCRAM_SHA_256,
+         ScramDigestPassword.ALGORITHM_SCRAM_SHA_384,
+         ScramDigestPassword.ALGORITHM_SCRAM_SHA_512,
+         DigestPassword.ALGORITHM_DIGEST_MD5,
+         DigestPassword.ALGORITHM_DIGEST_SHA,
+         DigestPassword.ALGORITHM_DIGEST_SHA_256,
+         DigestPassword.ALGORITHM_DIGEST_SHA_384,
+         DigestPassword.ALGORITHM_DIGEST_SHA_512
+   );
+
+   private final Path serverRoot;
+   private final Path usersFile;
+   private final Path groupsFile;
+   private Properties users = new Properties();
+   private Properties groups = new Properties();
+   private String realm = DEFAULT_REALM_NAME;
+   private Boolean plainText = false;
+
+   public UserTool(String serverRoot) {
+      this(serverRoot, DEFAULT_USERS_PROPERTIES_FILE, DEFAULT_GROUPS_PROPERTIES_FILE);
+   }
+
+   public UserTool(String serverRoot, String usersFile, String groupsFile) {
+      this(serverRoot != null ? Paths.get(serverRoot) : null,
+            usersFile != null ? Paths.get(usersFile) : null,
+            groupsFile != null ? Paths.get(groupsFile) : null);
+   }
+
+   public UserTool(Path serverRoot, Path usersFile, Path groupsFile) {
+      installSecurityProvider();
+      if (serverRoot != null && serverRoot.isAbsolute()) {
+         this.serverRoot = serverRoot;
+      } else {
+         String serverHome = System.getProperty("infinispan.server.home.path");
+         Path serverHomePath = serverHome == null ? Paths.get("") : Paths.get(serverHome);
+         if (serverRoot == null) {
+            this.serverRoot = serverHomePath.resolve("server");
+         } else {
+            this.serverRoot = serverHomePath.resolve(serverRoot);
+         }
+      }
+
+      if (usersFile == null) {
+         this.usersFile = this.serverRoot.resolve("conf").resolve(DEFAULT_USERS_PROPERTIES_FILE);
+      } else if (usersFile.isAbsolute()) {
+         this.usersFile = usersFile;
+      } else {
+         this.usersFile = this.serverRoot.resolve("conf").resolve(usersFile);
+      }
+      if (groupsFile == null) {
+         this.groupsFile = this.serverRoot.resolve("conf").resolve(DEFAULT_GROUPS_PROPERTIES_FILE);
+      } else if (groupsFile.isAbsolute()) {
+         this.groupsFile = groupsFile;
+      } else {
+         this.groupsFile = this.serverRoot.resolve("conf").resolve(groupsFile);
+      }
+      load();
+   }
+
+   private void installSecurityProvider() {
+      WildFlyElytronPasswordProvider instance = WildFlyElytronPasswordProvider.getInstance();
+      if (java.security.Security.getProvider(instance.getName()) == null) {
+         java.security.Security.insertProviderAt(instance, 1);
+      }
+   }
+
+   private void load() {
+      if (Files.exists(usersFile)) {
+         try (Reader reader = Files.newBufferedReader(usersFile)) {
+            users.load(reader);
+            // TODO: detect REALM_NAME and ALGORITHM
```

Comment:
No, I've made it so that you cannot add encrypted users to a clear-text password file and vice-versa. I've also added an `encrypt-all` subcommand which converts a clear-text file to encrypted in one fell swoop